src/pages/post/index.tsx

```diff
@@ -12,42 +12,53 @@ import {
 } from '@offer-ui/react'
 import type {
   ImageInfo,
-  // UploaderOnChangeHandler,
+  UploaderOnChangeHandler,
   SelectOnChangeHandler,
   InputProps
 } from '@offer-ui/react'
-import type { ReactElement, ChangeEventHandler, ChangeEvent } from 'react'
+import { useRouter } from 'next/router'
+import type { ReactElement, ChangeEventHandler } from 'react'
 import { useState } from 'react'
+import { useUploadImagesQuery } from '@apis/image/queries'
+import { useGetCategoriesQuery, useCreatePostMutation } from '@apis/post'
+import { localeCurrencyToNumber } from '@utils/format'
 import { PostForm } from '@components'
-import { CATEGORIES, TRADE_TYPES, PRODUCT_CONDITIONS } from '@constants'
+import { TRADE_TYPES, PRODUCT_CONDITIONS } from '@constants'
 import { useResponsive } from '@hooks'
+import type { ProductConditionCodes, TradeTypeCodes } from '@types'
 
 type PostFormStatus = {
-  imageUrls: ImageInfo[] | null
   title: string
-  category: number | null
-  tradeArea: string
-  productCondition: string
-  tradeMethod: number | null
-  content: string
-  price: number
+  imageInfos: ImageInfo[]
+  category: string
+  price: string
+  location: string
+  productCondition: ProductConditionCodes | ''
+  tradeType: TradeTypeCodes | ''
+  description: string
 }
 
 type HandleUpdatePostForm = ChangeEventHandler<
   HTMLTextAreaElement | HTMLInputElement | HTMLFormElement
 >
+const initialPostForm: PostFormStatus = {
+  imageInfos: [],
+  title: '',
+  category: '',
+  location: '',
+  productCondition: '',
+  tradeType: '',
+  description: '',
+  price: ''
+}
 
 const PostPage = (): ReactElement => {
-  const [postForm, setPostForm] = useState<PostFormStatus>({
-    imageUrls: null,
-    title: '',
-    category: null,
-    tradeArea: '',
-    productCondition: '',
-    tradeMethod: null,
-    content: '',
-    price: 0
-  })
+  const { mutateAsync: createPost } = useCreatePostMutation()
+  const { mutateAsync: postUploadImages } = useUploadImagesQuery()
+  const { data: categoriesData } = useGetCategoriesQuery()
```

Comment:
```suggestion
  const createPost = useCreatePostMutation()
  const postUploadImages = useUploadImagesQuery()
  const categories= useGetCategoriesQuery()
```

query랑 mutation이 많아져서 구조 분해 할당하지 않고 변수에서 꺼내서 사용하기로 했던 것 같습니다.!

Reply:
[변경완료](https://github.com/price-offer/offer-fe/pull/179/commits/17abbec909f742c99514d4e2c6f59e8da13e77e8) !